### PR TITLE
Parsing environment variables bug in kedro azureml run

### PR DESCRIPTION
**Pull request by datboi2001**
Hello,
I noticed that when I used the command `kedro azureml run --extra-env`, the cli fails to parse correct environment variable when the value has a `=` somewhere in the string. For example, if the key value pair is `ACCOUNT_KEY=CDEF=123`, then the value will become `CDEF`, instead of `CDEF=123`. 
I have fixed the bug in `cli_functions.py` in function `parse_extra_env_params` and added a new test case in `test_cli.py`

### CODE REVIEW

File: kedro_azureml/cli_functions.py [hunk 1]
```diff
@@ -147,4 +147,4 @@ def parse_extra_env_params(extra_env):
         if not re.match("[A-Za-z0-9_]+=.*", entry):
             raise Exception(f"Invalid env-var: {entry}, expected format: KEY=VALUE")
 
-    return {(e := entry.split("="))[0]: e[1] for entry in extra_env}
+    return {(e := entry.split("="))[0]: '='.join(e[1]) for entry in extra_env}
```
Comment (fdroessler) [inline kedro_azureml/cli_functions.py:150]: nice catch, what do you think about
`return {(e := entry.split("=", maxsplit=1))[0]: e[1] for entry in extra_env}`?

Comment (datboi2001) [reply kedro_azureml/cli_functions.py:150]: Great idea, will fix it

**Comment by marrrcin** [comment]
Nice catch @datboi2001 !

**Comment by marrrcin** [comment]
Unit tests are failing, please run pre-commit and submit the changes.

**Comment by marrrcin** [comment]
Pre-commits are still not passing (black) @datboi2001

**Comment by datboi2001** [comment]
@marrrcin, I removed the unit test that I wrote. Hopefully, it will work this time

**Comment by fdroessler** [comment]
I think you should keep the test in! I think what @marrrcin meant was that the linting checks are failing. The best way to make sure all of these are passing is to run `pre-commit run --all` and `pytest .` to ensure that both all the formatting is done correctly and all tests are passing :) Please note that sometimes running `pre-commit run --all` will result in file changes that need to be added to the staged files via `git add` before doing the commit. :)

**Comment by datboi2001** [comment]
@fdroessler thanks for the help! I ran all of the commands you suggested and have pushed the reformatted file.

**Comment by marrrcin** [comment]
Merged, thanks for your contribution @datboi2001 !

**Pull request merged**